core/src/main/java/org/axonframework/eventsourcing/CachingEventSourcingRepository.java

```diff
@@ -89,10 +142,31 @@ public CachingEventSourcingRepository(AggregateFactory<T> aggregateFactory, Even
     public CachingEventSourcingRepository(AggregateFactory<T> aggregateFactory, EventStore eventStore,
                                           LockFactory lockFactory, Cache cache,
                                           SnapshotTriggerDefinition snapshotTriggerDefinition) {
-        super(aggregateFactory, eventStore, lockFactory, snapshotTriggerDefinition);
+        this(aggregateFactory, eventStore, lockFactory, cache, snapshotTriggerDefinition, null);
+    }
+
+    /**
+     * Initializes a repository with a the given {@code aggregateFactory} and a pessimistic locking strategy.
+     * <p>
+     * Note that an optimistic locking strategy is not compatible with caching.
+     *
+     * @param aggregateFactory          The factory for new aggregate instances
+     * @param eventStore                The event store that holds the event streams for this repository
+     * @param lockFactory               The lock factory restricting concurrent access to aggregate instances
+     * @param cache                     The cache in which entries will be stored
+     * @param snapshotTriggerDefinition The definition describing when to trigger a snapshot  @see
+     *                                  LockingRepository#LockingRepository(Class)
+     * @param repositoryProvider        Provides repositories for given aggregate types
```

Comment:
So I already commented on this line of javadoc a couple of files earlier, but I just now notice it is reoccurring, a lot.
Which is fine of course, but I just like to point out that if my earlier comment of 'is the use of `given` okay in this javadoc line?' is correct, that it should be adjusted everywhere where you've placed that line.